File: scratch/check_blade.py

import re

def check_blade_directives(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find all @if, @elseif, @else, @endif
    directives = re.findall(r'(@if|@elseif|@else|@endif)', content)
    
    level = 0
    mismatches = []
    
    for i, d in enumerate(directives):
        if d == '@if':
            level += 1
        elif d == '@endif':
            level -= 1
            if level < 0:
                mismatches.append(f"Unexpected @endif at index {i}")
                level = 0
        elif d == '@else' or d == '@elseif':
            if level == 0:
                mismatches.append(f"Unexpected {d} at index {i} (level 0)")
    
    if level > 0:
        mismatches.append(f"Unclosed @if (level {level} at end)")
        
    return mismatches, directives

mismatches, directives = check_blade_directives(r'c:\Users\jcsdi\Documents\Driving School Management System\DrivingApp\resources\views\school\guest\dashboard.blade.php')
print(f"Mismatches: {mismatches}")
print(f"Directives: {directives}")

File: scratch/check_blade.php
<?php
$filepath = 'c:\Users\jcsdi\Documents\Driving School Management System\DrivingApp\resources\views\school\guest\dashboard.blade.php';
$content = file_get_contents($filepath);

preg_match_all('/(@if|@elseif|@else|@endif)/', $content, $matches);
$directives = $matches[0];

$level = 0;
$mismatches = [];

foreach ($directives as $i => $d) {
    if ($d === '@if') {
        $level++;
    } elseif ($d === '@endif') {
        $level--;
        if ($level < 0) {
            $mismatches[] = "Unexpected @endif at index $i";
            $level = 0;
        }
    } elseif ($d === '@else' || $d === '@elseif') {
        if ($level === 0) {
            $mismatches[] = "Unexpected $d at index $i (level 0)";
        }
    }
}

if ($level > 0) {
    $mismatches[] = "Unclosed @if (level $level at end)";
}

echo "Mismatches: " . implode(", ", $mismatches) . "\n";
echo "Directives count: " . count($directives) . "\n";
foreach ($directives as $i => $d) {
    echo "$i: $d\n";
}
?>
